File: rust/src/models/subscription_add_on_price_override.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::price_entry::PriceEntry;

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct SubscriptionAddOnPriceOverride {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub name: Option<String>,

    pub price_entry: PriceEntry,
}

impl SubscriptionAddOnPriceOverride {
    pub fn new(price_entry: PriceEntry) -> Self {
        Self {
            name: None,
            price_entry,
        }
    }
}

File: typescript/src/models/subscriptionAddOnPriceOverride.ts
// this file is @generated
import { type PriceEntry, PriceEntrySerializer } from "./priceEntry";

export interface SubscriptionAddOnPriceOverride {
  name?: string | null;

  priceEntry: PriceEntry;
}

export const SubscriptionAddOnPriceOverrideSerializer = {
  _fromJsonObject(object: any): SubscriptionAddOnPriceOverride {
    return {
      name: object["name"],
      priceEntry: PriceEntrySerializer._fromJsonObject(object["price_entry"]),
    };
  },

  _toJsonObject(self: SubscriptionAddOnPriceOverride): any {
    return {
      name: self.name,
      price_entry: PriceEntrySerializer._toJsonObject(self.priceEntry),
    };
  },
};
